chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,289 @@
+import numba
+import numpy as np
+import requests.exceptions
+from astropy.table import Table
+from cheta import fetch_sci
+from cxotime import CxoTimeLike
+from mica.archive import aca_l0
+from mica.archive.aca_dark import get_dark_cal_props
+from ska_helpers import retry
+
+from chandra_aca import maude_decom
+from chandra_aca.dark_model import dark_temp_scale_img
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def get_tccd_data(start: CxoTimeLike, stop: CxoTimeLike, source="maude"):
+    """
+    Get the CCD temperature for a given time range.
+
+    This is a light wrapper around cheta.fetch_sci.Msid("aacccdpt", start, stop)
+    to handle an option to use the maude data source in an explicit way.
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start time of the time range.
+    stop : CxoTimeLike
+        Stop time of the time range.
+    source : str, optional
+        Source of the CCD temperature data. If 'maude', override the fetch_sci data source
+        to 'maude allow_subset=False'. Else, use the cheta default.
+
+    Returns
+    -------
+    t_ccd_vals : np.array
+        CCD temperature values.
+    t_ccd_times : np.array
+        Times corresponding to the CCD temperature values.
+    """
+    if source == "maude":
+        # Override the data_source to be explicit about maude source.
+        with fetch_sci.data_source("maude allow_subset=False"):
+            dat = fetch_sci.Msid("aacccdpt", start, stop)
+    else:
+        dat = fetch_sci.Msid("aacccdpt", start, stop)
+    return dat.vals, dat.times
+
+
+@retry.retry(exceptions=requests.exceptions.RequestException, delay=5, tries=3)
+def _get_dcsub_aca_images(
+    start: CxoTimeLike = None,
+    stop: CxoTimeLike = None,
+    aca_image_table=None,
+    t_ccd_vals=None,
+    t_ccd_times=None,
+    img_dark=None,
+    tccd_dark=None,
+    source="maude",
+):
+    """
+    Get dark current subtracted ACA images for a given time range.
+
+    This private method has expanded options for unit testing.
+    Users should use get_dcsub_aca_images instead.
```

Comment:
I would very much like to have the ability to use these options!

While I understand you made this for unit testing, it also has a use-case within aca_view. In aca_view, data can come from several places, and functions that hide the data fetching, and do not allow to override it, are kind of a pain.

As an example of how this would be used inside aca_view, I made this gist: https://gist.github.com/javierggt/7181af2a737064d3658a70d62396e96a

Reply:
It would be nice to use this within the agasc supplement code too, but I have not looked how it would fit.